loop.c

```diff
@@ -104,12 +105,18 @@ void loop_poll(struct loop *loop) {
 		clock_gettime(CLOCK_MONOTONIC, &now);
 		struct loop_timer *timer = NULL, *tmp_timer = NULL;
 		wl_list_for_each_safe(timer, tmp_timer, &loop->timers, link) {
+			if (timer->removed) {
+				wl_list_remove(&timer->link);
+				free(timer);
```

Comment:
We should add a `continue` statement here to avoid use-after-free

Reply:
Haha, I should've caught that. Sorry